velox/expression/tests/ExprTest.cpp

```diff
@@ -2059,6 +2072,109 @@ TEST_F(ExprTest, constantToString) {
       exprSet.exprs()[2]->toString());
 }
 
+TEST_F(ExprTest, constantToSql) {
+  auto toSql = [&](const variant& value) {
+    exec::ExprSet exprSet({makeConstantExpr(value)}, execCtx_.get());
+    return exprSet.expr(0)->toSql();
+  };
+
+  ASSERT_EQ(toSql(true), "TRUE");
+  ASSERT_EQ(toSql(false), "FALSE");
+  ASSERT_EQ(toSql(variant::null(TypeKind::BOOLEAN)), "NULL");
+
+  ASSERT_EQ(toSql((int8_t)23), "23::TINYINT");
+  ASSERT_EQ(toSql(variant::null(TypeKind::TINYINT)), "NULL");
+
+  ASSERT_EQ(toSql((int16_t)23), "23::SMALLINT");
+  ASSERT_EQ(toSql(variant::null(TypeKind::SMALLINT)), "NULL");
+
+  ASSERT_EQ(toSql(23), "23::INTEGER");
+  ASSERT_EQ(toSql(variant::null(TypeKind::INTEGER)), "NULL");
+
+  ASSERT_EQ(toSql(2134456LL), "2134456::BIGINT");
+  ASSERT_EQ(toSql(variant::null(TypeKind::BIGINT)), "NULL");
+
+  ASSERT_EQ(toSql(1.5f), "1.5::REAL");
+  ASSERT_EQ(toSql(variant::null(TypeKind::REAL)), "NULL");
+
+  ASSERT_EQ(toSql(-78.456), "-78.456::DOUBLE");
+  ASSERT_EQ(toSql(variant::null(TypeKind::DOUBLE)), "NULL");
+
+  ASSERT_EQ(toSql("This is a test."), "'This is a test.'");
+  ASSERT_EQ(
+      toSql("This is a \'test\' with single quotes."),
+      "'This is a \'\'test\'\' with single quotes.'");
+  ASSERT_EQ(toSql(variant::null(TypeKind::VARCHAR)), "NULL");
+
+  auto toSqlComplex = [&](const VectorPtr& vector) {
+    exec::ExprSet exprSet({makeConstantExpr(vector, 0)}, execCtx_.get());
+    return exprSet.expr(0)->toSql();
+  };
+
+  ASSERT_EQ(
+      toSqlComplex(makeArrayVector<int32_t>({{1, 2, 3}})),
+      "ARRAY[1::INTEGER, 2::INTEGER, 3::INTEGER]");
+  ASSERT_EQ(toSql(variant::null(TypeKind::ARRAY)), "NULL");
+
+  ASSERT_EQ(
+      toSqlComplex(
+          makeMapVector<int32_t, int32_t>({{{1, 10}, {2, 20}, {3, 30}}})),
+      "map(ARRAY[1::INTEGER, 2::INTEGER, 3::INTEGER], ARRAY[10::INTEGER, 20::INTEGER, 30::INTEGER])");
+  ASSERT_EQ(
+      toSqlComplex(BaseVector::createNullConstant(
+          MAP(INTEGER(), VARCHAR()), 10, pool())),
+      "NULL");
+
+  ASSERT_EQ(
+      toSqlComplex(makeRowVector({
+          makeFlatVector<int32_t>({1, 2, 3}),
+          makeFlatVector<bool>({true, false, true}),
+      })),
+      "row_constructor(1::INTEGER, TRUE)");
+  ASSERT_EQ(
+      toSqlComplex(BaseVector::createNullConstant(
+          ROW({"a", "b"}, {BOOLEAN(), DOUBLE()}), 10, pool())),
+      "NULL");
+}
+
+TEST_F(ExprTest, toSql) {
+  auto rowType = ROW({"a", "b", "c.d"}, {INTEGER(), BIGINT(), VARCHAR()});
+
+  // CAST.
+  testToSql("a + 3", rowType);
+  testToSql("a * b", rowType);
+  testToSql("a * 1.5", rowType);
+
+  // SWITCH..
+  testToSql("if(a > 0, 1, 10)", rowType);
+  testToSql("if(a = 10, true, false)", rowType);
+  testToSql("case a when 7 then 1 when 11 then 2 else 0 end", rowType);
```

Comment:
Can we have an expression instead of just constants in the then or when clauses ? Would be great to add to this test.